cmake/FindOpenSSL.cmake

```diff
@@ -1,136 +0,0 @@
-include( FindPackageHandleStandardArgs )
-
-set( OPENSSL_GOOD_VERSION TRUE )
-if( OPENSSL_INCLUDE_DIR AND OPENSSL_LIBRARIES )
-  set( OPENSSL_FOUND TRUE )
-else()
-  find_path(
-    OPENSSL_INCLUDE_DIR
-    NAMES openssl/ssl.h
-    HINTS
-    ${OPENSSL_ROOT_DIR}
-    PATH_SUFFIXES
-    include )
-
-  file(STRINGS ${OPENSSL_INCLUDE_DIR}/openssl/opensslv.h opensslvers REGEX "^# define OPENSSL_VERSION_NUMBER")
-  string(REGEX REPLACE "# define[ ]+OPENSSL_VERSION_NUMBER[ ]+" "" opensslvers ${opensslvers})
-
-  set( LIBSSLNAME ssl )
-  set( LIBCRYPTONAME crypto )
-
-  if ( OPENSSL_GOOD_VERSION )
-
-    find_library(
-      OPENSSL_SSL_LIBRARY
-      NAMES ${LIBSSLNAME}
-      HINTS
-      ${OPENSSL_ROOT_DIR}
-      PATH_SUFFIXES
-      ${LIBRARY_PATH_PREFIX}
-      ${LIB_SEARCH_OPTIONS})
-
-    string(FIND ${OPENSSL_SSL_LIBRARY} ".a" hasstaticext)
-    if (NOT ${hasstaticext} EQUAL -1)
-      if ( ${opensslvers} LESS "0x1000201fL" )
-        set( OPENSSL_GOOD_VERSION FALSE )
-        message(WARNING " >>> Cannot build XRootD crypto modules: static openssl build version is < 1.0.2")
-      else()
-        set(OPENSSL_USE_STATIC TRUE)
-        set( LIBCRYPTONAME libcrypto.a )
-      endif()
-      message("-- Using OpenSSL static libraries (version: " ${opensslvers} ")")
-    endif()
-
-  endif()
-
-  if ( OPENSSL_GOOD_VERSION )
-    find_library(
-      OPENSSL_CRYPTO_LIBRARY
-      NAMES ${LIBCRYPTONAME}
-      HINTS
-      ${OPENSSL_ROOT_DIR}
-      PATH_SUFFIXES
-      ${LIBRARY_PATH_PREFIX}
-      ${LIB_SEARCH_OPTIONS})
-
-    set( OPENSSL_LIBRARIES ${OPENSSL_SSL_LIBRARY} ${OPENSSL_CRYPTO_LIBRARY} )
-
-    find_package_handle_standard_args(
-      OpenSSL
-      DEFAULT_MSG
-      OPENSSL_LIBRARIES)
-
-    mark_as_advanced( OPENSSL_INCLUDE_DIR OPENSSL_LIBRARIES )
-
-  endif()
-endif()
-
-
-#-------------------------------------------------------------------------------
-# Check for the TLS support in the OpenSSL version that is available
-# (assume available if use of static libs is detected and the openssl version
-# is at least 1.0.2)
-#-------------------------------------------------------------------------------
-
-if ( OPENSSL_FOUND )
-
-  if( OPENSSL_USE_STATIC AND OPENSSL_GOOD_VERSION )
-
-    add_definitions( -DHAVE_TLS -DHAVE_TLS12 -DHAVE_TLS11 -DHAVE_TLS1 -DHAVE_DH_PADDED -DHAVE_DH_PADDED_FUNC )
```

Comment:
I'm not sure if removing this file is the right thing to do, those definitions are used in the source code.

Reply:
I grepped for them, and most are not used. The one that was used I added back (`-DHAVE_DH_PADDED`).

Reply:
I think when CMake provides a find module, it's better to use the upstream one, unless the version of CMake we support doesn't provide it. Usually find modules should not change compile options of the projects using them, but just export which compile options may be needed as variables, etc. If we need to `add_definitions(...)` for things specific to XRootD, then we should do outside of the find module, like was already being done in `cmake/XRootDFindLibs.cmake`. At least for `-DHAVE_TLS`, the definition added by the find module was redundant.

Reply:
from what I see `HAVE_TLS` is being used in XrdTls/XrdTlsContext.cc

Reply:
Ah, indeed, `cmake/XRootDFindLibs.cmake` has `HAVE_SSL`, not `HAVE_TLS`. I mistook one for the other, too many Ss and Ls. I will add also `HAVE_TLS`, I did see that when I grepped.

Reply:
This is addressed in my last commit.